uswds/uswds-next
# USWDS-Next - Storybook: Add initial implementation

Closes #6.

Blocked by #8.

Initial setup of StorybookJS in repo. Includes a few examples for:

- Custom docs with MDX.
- Embedding existing MD content.
- Autogenerated docs.
- Component previews.

Also created a small custom theme called `UswdsTheme.js` to use Public Sans, bump up default font size, and use some USWDS colors.

**Welcome page**
Includes an example of what custom documentation using [Storybook blocks](https://storybook.js.org/docs/writing-docs/doc-blocks) looks like. 
![image](https://github.com/uswds/uswds-next/assets/3385219/3baaa697-f6e6-4912-8ac3-0d4a4ab13375)

**Embedded README**
Example of embedding an existing README into page. Could be used for CHANGELOGs.
![image](https://github.com/uswds/uswds-next/assets/3385219/22fef337-7f80-45bc-adb2-bc16d6770796)

**Component docs**
Autogenerated component docs.
![image](https://github.com/uswds/uswds-next/assets/3385219/63cb2882-150c-443f-8818-4decca670297)

**Component preview**
Example of importing Lit component with configurable props.
![image](https://github.com/uswds/uswds-next/assets/3385219/d31c2fdb-c2e0-40a6-bce3-e5a92b2c04fe)

## How to test

1. Go into `web-components` directory.
2. Run `npm install`.
3. Run `npm start`.

You should have a working storybook library powered by Vite.
